### PR TITLE
[BPK-3718] Update publishing tasks

### PR DESCRIPTION
**Pull request by tiagoengel**
+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-react-native/blob/master/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: android/gradle-maven-push.gradle [hunk 1]
```diff
@@ -24,11 +24,17 @@ afterEvaluate { project ->
     }
 
     project.ext.getJFrogPassword = {
-        return System.getenv('JFROG_PASSWORD') ?: localProps.getProperty("jfrog_password")
+        if (project.hasProperty("SKYSCANNER_ARTIFACTORY_MAVEN_PASSWORD")) {
```
Comment (tiagoengel) [inline android/gradle-maven-push.gradle]: This can be set now using the mshell login client

File: package.json [hunk 2]
```diff
@@ -32,10 +32,7 @@
     "fix-bpk-dependencies": "",
     "test": "npm run lint && npm run check-bpk-dependencies && jest --coverage && npm run flow-check-ios && npm run flow-check-android && npm run spellcheck",
     "check-owners": "npm whoami && ( node scripts/npm/get-owners.js | grep -E $(npm whoami) ) && node scripts/npm/check-owners.js",
-    "check-maven-auth": "(cd android && ./gradlew :react-native-bpk-component-calendar:checkMavenCredentials)",
-    "publish": "npm run check-maven-auth && npm run check-owners && git pull --rebase && npm run flow stop && npm test && npm run publish:prepare && lerna publish && npm run publish:bridges",
-    "publish:prepare": "node scripts/npm/native-bridges.js prepare",
-    "publish:bridges": "node scripts/npm/native-bridges.js publish",
+    "publish": "npm run check-owners && git pull --rebase && npm run flow stop && npm test && node scripts/release.js",
```
Comment (k0nserv) [inline package.json]: A couple of notes on this from the iOS repository which we have learned over a bunch of failed releases:

+ Check that the branch is [master](https://github.com/Skyscanner/backpack-ios/blob/master/Rakefile#L122)
+ Check [publish rights](https://github.com/Skyscanner/backpack-ios/blob/master/Rakefile#L123), I guess `check-owners` already does that
+ Ensure git branch is [up to date with remote](https://github.com/Skyscanner/backpack-ios/blob/master/Rakefile#L124) via [`branc_up_to_date`](https://github.com/Skyscanner/backpack-ios/blob/458e5527b5ee39d790205c002e51b93876496820/Rakefile#L42-L45)
+ Run a [git fetch](https://github.com/Skyscanner/backpack-ios/blob/458e5527b5ee39d790205c002e51b93876496820/Rakefile#L83) before starting the release process. Doing a pull will not update tags and at least on iOS it's important that you have the latest tagged release available
+ Ensure the repo is [pristine](https://github.com/Skyscanner/backpack-ios/blob/458e5527b5ee39d790205c002e51b93876496820/Rakefile#L128) after running any build steps.

File: scripts/release.js [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+ * @example
+ * ```js
+ * gradleExec(['tasks'], process.stdout, process.stderr)
+ * ```
+ *
+ * @param {Array<String>} cmd the command
+ * @param {WritableStream} stdout stream to redirect standard output
+ * @param {WritableStream} stderr stream to redirect error output
+ *
+ * @returns {Promise} a promise object
```
Comment (k0nserv) [inline scripts/release.js]: Top notch documentation 👌

File: scripts/release.js [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+ */
+function prepareRelease() {
+  console.log('🧹   ', '> Cleaning up');
+  execSync(`rm -rf ${distRoot} 2> /dev/null`);
+  execSync(`mkdir ${distRoot}`);
+
+  console.log('📦  ', '> Preparing package');
+  // cp -r src/* is much faster than copying file by file,
+  // to filter out files you don't want to publish edit `.npmignore`.
+  execSync(`cp -r ${srcRoot}/* ${distRoot}`);
```
Comment (k0nserv) [inline scripts/release.js]: Should this be abstracted into a `npm run build` task?

File: package.json (hunk 2 above)
Comment (tiagoengel) [reply package.json]: Good points! About making sure it's pristine `release-it` does it for us via `requireCleanWorkingDir: true`, I'll look into adding the others

File: scripts/release.js (hunk 4 above)
Comment (tiagoengel) [reply scripts/release.js]: Yeah, good point. Will be better to test it.

File: package.json (hunk 2 above)
Comment (tiagoengel) [reply package.json]: I've added all the checks, the branch check is done by the lib as well.

File: scripts/release.js (hunk 4 above)
Comment (tiagoengel) [reply scripts/release.js]: Done

**Pull request merged**